Tests/HubTests/HubApiTests.swift

```diff
@@ -0,0 +1,141 @@
+                    ])
+                )
+            }()
+
+            // Glob patterns are case sensitive
+            try await {
+                let filenames = try await Hub.getFilenames(from: "coreml-projects/Llama-2-7b-chat-coreml", matching: "*.JSON")
+                XCTAssertEqual(
+                    filenames,
+                    []
+                )
+            }()
+        } catch {
+            XCTFail("\(error)")
+        }
+    }
+
+    func testFilenameRetrievalFromDirectories() async {
+        do {
+            // Contents of all directories matching a pattern
+            let filenames = try await Hub.getFilenames(from: "coreml-projects/Llama-2-7b-chat-coreml", matching: "*.mlpackage/*")
+            XCTAssertEqual(
+                Set(filenames),
+                Set([
+                    "llama-2-7b-chat.mlpackage/Manifest.json",
+                    "llama-2-7b-chat.mlpackage/Data/com.apple.CoreML/FeatureDescriptions.json",
+                    "llama-2-7b-chat.mlpackage/Data/com.apple.CoreML/Metadata.json",
+                    "llama-2-7b-chat.mlpackage/Data/com.apple.CoreML/model.mlmodel",
+                    "llama-2-7b-chat.mlpackage/Data/com.apple.CoreML/weights/weight.bin",
+
+                ])
+            )
+        } catch {
+            XCTFail("\(error)")
+        }
+    }
+}
+
+class SnapshotDownloadTests: XCTestCase {
+    let repo = "coreml-projects/Llama-2-7b-chat-coreml"
+    let downloadDestination: URL = {
+        let base = FileManager.default.urls(for: .cachesDirectory, in: .userDomainMask).first!
+        return base.appending(component: "huggingface-tests")
+    }()
+    
+    override func setUp() {
+    }
+
+    override func tearDown() {
+        do {
+            try FileManager.default.removeItem(at: downloadDestination)
+        } catch {
+            print("Can't remove test download destination \(downloadDestination), error: \(error)")
+        }
+    }
+
+    func getRelativeFiles(url: URL) -> [String] {
+        var filenames: [String] = []
+        let prefix = downloadDestination.appending(path: "models/\(repo)").path.appending("/")
+
+        if let enumerator = FileManager.default.enumerator(at: url, includingPropertiesForKeys: [.isRegularFileKey], options: [.skipsHiddenFiles], errorHandler: nil) {
+            for case let fileURL as URL in enumerator {
+                do {
+                    let resourceValues = try fileURL.resourceValues(forKeys: [.isRegularFileKey])
+                    if resourceValues.isRegularFile == true {
+                        filenames.append(String(fileURL.path.suffix(from: prefix.endIndex)))
+                    }
+                } catch {
+                    print("Error reading file resources: \(error)")
+                }
+            }
+        }
+        return filenames
+    }
+    
+    func testDownload() async throws {
+        let hubApi = HubApi(downloadBase: downloadDestination)
```

Comment:
This is how to create a downloader with a specific local filesystem destination.